src/middlewared/middlewared/plugins/jail.py

```diff
@@ -1094,7 +1062,7 @@ def get_local_plugin_version(self, plugin, index_json, iocroot):
 
         try:
             base_plugin = plugin.rsplit('_', 1)[0]  # May have multiple
-            primary_pkg = index_json[base_plugin]['primary_pkg']
+            primary_pkg = index_json[base_plugin].get('primary_pkg') or plugin
```

Comment:
Perhaps use the first pkg in the pkgs list?

Reply:
That causes issues for plugins which don't have `primary_pkg` key defined. Let's take asigra, it will start showing up version number for asigra based on `ca_root_nss` pkg which is the first. Reliability is hurt this way